data-dictionary.md

```diff
@@ -342,17 +586,30 @@ Timestamps are expected to use the OData [edm:DateTimeOffset](http://docs.oasis-
     Then "ModificationTimestamp" MUST be "Timestamp" data type
 ```
 
-### Lookups Resource and Location-Based Fields
+### Lookup Resource
+RESO supports use of a Data Dictionary resource in order to advertise lookup metadata. This has the advantage of providing human friendly lookup values as well as the ability to more easily replicate large sets of enumerations, such as subdivisions or cities.
+
+The following testing rules will be used during RESO Certification:
+* Check the data type of the lookup field, e.g. `StandardStatus` or `AccessibilityFeatures`, it should be `Edm.String` for single enumerations and `Collection(Edm.String)` for multiple enumerations. 
+* Check that the required annotation is present in the OData XML metadata from the server, and in the correct format.
+* Check that the `Lookup` resource is present in the metadata, exists on the server, and is defined correctly.
+* All records will be replicated from the `Lookup` resource using `$top` and `$skip` queries. Payload data will then be correlated with what has been advertised to ensure that enumerations are present for each annotated lookup name. 
+
+Servers **MUST** be able to provide the entire set of lookups relevant for testing through the replication operation so, for example, if a given system has 101 records from the `$count=true` query option but only 100 records were fetched, this would fail. The opposite is also true, if 100 records were advertised and 101 were found, this would not pass testing.
 
-After further discussion in the Transport and Certification subgroups, certain lookups will allowed to be Edm.String for String List, Single, or Collection(Edm.String) for String List, Multi, pending further discussion of Lookups, as outlined in [RCP-032](https://members.reso.org/display/RESOWebAPIRCP/RCP+-+WEBAPI-032++Lookups+Resource+for+Enumeration+Metadata). 
+See the [Lookup resource section](#section-22-lookup-resource-for-enumeration-metadata) in the specification for more information.
 
-Note that RCP-032 started off as being specifically for location-based lookups, such as City and CountyOrParish, and has potentially been expanded to include any Lookup. 
+### Schema Validation
+RESO Certification will include strict schema validation to ensure that the data available in the payload matches what's advertised in the metadata.
+
+This will consist of creating a JSON Schema representation of the available metadata, including the Field and Lookup resources, and validating all responses from the server with it. [JSON Schema allows](http://json-schema.org/understanding-json-schema/reference/object.html#additional-properties) an `additionalProperties` flag to be set to `false`, meaning that if any additional properties or enumerated values exist in the payload that aren't in the metadata, schema validation will fail. 
```

Comment:
> if any additional properties or enumerated values exist in the payload that aren't in the metadata, schema validation will fail

I support this for enumerated values, but this might be stricter than OData with respect to additional properties and I don't know if it's a necssary rule.

Reply:
We can generate JSON schema from the OData metadata we consume. I'd been testing something similar for non-OData APIs since the payload is the same and we also want it for that case.

In Data Dictionary and Payloads 2.0, anything that's not advertised in the metadata won't pass testing per prior workgroup discussions so in that case additionalProperties would be set to `false` and we would have had to consume all field and enumeration definitions from the available metadata, including the Field and Lookup resources.